/*
 Copyright 2023 The Kapacity Authors.

 Licensed under the Apache License, Version 2.0 (the "License");
 you may not use this file except in compliance with the License.
 You may obtain a copy of the License at

     http://www.apache.org/licenses/LICENSE-2.0

 Unless required by applicable law or agreed to in writing, software
 distributed under the License is distributed on an "AS IS" BASIS,
 WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
*/

syntax = "proto3";
import "metric.proto";
import "google/protobuf/timestamp.proto";
import "google/protobuf/duration.proto";

package io.kapacitystack.metric;
option go_package = "github.com/traas-stack/kapacity/pkg/metric/service/api";

service ProviderService {
  rpc QueryLatest (QueryLatestRequest) returns (QueryLatestResponse);
  rpc Query (QueryRequest) returns (QueryResponse);
}

message QueryLatestRequest {
  Query query = 1;
}

message QueryLatestResponse {
  repeated Sample samples = 1;
}

message QueryRequest {
  Query query = 1;
  google.protobuf.Timestamp start = 2;
  google.protobuf.Timestamp end = 3;
  google.protobuf.Duration step = 4;
}

message QueryResponse {
  repeated Series series = 1;
}
